ports/stm32/machine_adc.c

```diff
@@ -330,11 +335,17 @@ void adc_config(ADC_TypeDef *adc, uint32_t bits) {
     uint32_t cfgr1 = res << ADC_CFGR1_RES_Pos;
     adc->CFGR1 = (adc->CFGR1 & ~cfgr1_clr) | cfgr1;
 
+    #elif defined(STM32U5)
+
+    uint32_t cfgr1_clr = ADC_CFGR1_CONT | ADC_CFGR1_EXTEN | ADC_CFGR1_RES;
```

Comment:
It looks like this is the same as the N6 code just above, so can be combined with it.